code/weapon/weapons.cpp

```diff
@@ -7497,7 +7524,55 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		weapon_hit_do_sound(other_obj, wip, hitpos, armed_weapon, quadrant);
 	}
 
-	if (wip->impact_weapon_expl_effect.isValid() && armed_weapon) {
+
+ 	bool valid_conditional_impact = false;
+	int relevant_armor_idx = -1;
+	float relevant_fraction = 1.0f;
+	float hit_angle = 0.0f;
+	vec3d reverse_incoming = (vmd_zero_vector - weapon_obj->orient.vec.fvec);
```

Comment:
The unary operator (-) works on vectors! So you can write `vec3d reverse_incoming = -weapon_obj->orient.vec.fvec` which should be a little clearer.

Reply:
According to VS, I can't.

`no operator "-" matches these operands`
`unary '-': 'vec3d' does not define this operator or a conversion to a type acceptable to the predefined operator`
`built-in operator '-' cannot be applied to an operand of type 'vec3d'`

Reply:
My apologies, too much time working with pof tools where this is the case. Either way the convention is `vm_vec_negate()`
```
vec3d reverse_incoming = weapon_obj->orient.vec.fvec;
vm_vec_negate(&reverse_incoming);
```
It is a little more verbose this way, and this is verging on pedantic, but is more consistent with the rest of the codebase and more explicit.